### PR TITLE
[Enhancement] Support large json in image

### PR DESCRIPTION
**Pull request by gengjun-git**
## Why I'm doing:
Currently, JSON serialization supports a maximum of 2G, because serialization converts the object into a JSON string first, and then writes the string into the image file, and the maximum size of a string in Java is only 2G.

## What I'm doing:
To support large json string, we can use the stream api of GSON: `JsonWriter` and `JsonReader`. 
Since we cannot know the length of the string in advance, so a new image file format is introduced: a json string must begin with "{" or "[", so primitive type is not supported, and there is nothing between the json string. To support rollback, we use a double write solution. The dir format is as follows
```
meta/image
         image.111
         /v2
              image.111
              checksum.111
```
`meta/image/image.111` is the old format image file, and `meta/image/v2/image.111` is the new format image file. Since the JsonReader will cache some chars in advance, so we cannot calculate the checksum of the specific meta-block, and the checksum is put into another file `checksum.111`;

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/persist/metablock/SRMetaBlockWriterV2.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        }
+        this.header = new SRMetaBlockHeader(id, numJson);
+        this.jsonWriter = jsonWriter;
+        this.numJsonWritten = 0;
+    }
+
+    @Override
+    public void writeJson(Object object) throws IOException, SRMetaBlockException {
+        if (isBasicType(object)) {
+            throw new SRMetaBlockException("can not write primitive type");
```
[the remark this answers is not in the document]
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/persist/metablock/SRMetaBlockWriterV2.java:39]: Two adjacent basic types will be considered as one when reading.

File: fe/fe-core/src/main/java/com/starrocks/persist/ImageLoader.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            return new SRMetaBlockReaderV2(jsonReader);
+        }
+    }
+
+    public void checkCheckSum() throws IOException {
+        if (imageFormatVersion == ImageFormatVersion.v2) {
+            Path checksumPath = Path.of(imageDir, "v2", Storage.CHECKSUM + "." + imageJournalId);
+            File checksumFile = new File(checksumPath.toUri());
+            if (!checksumFile.exists()) {
+                return;
```
[the remark this answers is not in the document]
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/persist/ImageLoader.java]: done

File: fe/fe-core/src/main/java/com/starrocks/persist/ImageLoader.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            }
+
+            long expectedCheckSum = Long.parseLong(Files.readString(checksumPath));
+            try (CheckedInputStream in = new CheckedInputStream(new FileInputStream(imageFile), new CRC32())) {
+                byte[] bytes = new byte[8192];
+                while (in.read(bytes) != -1) {
+                }
+
+                long realCheckSum = in.getChecksum().getValue();
```
[the remark this answers is not in the document]
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/persist/ImageLoader.java]: Yes, I use the checkedInputStream to judge the checksum at the end.

File: fe/fe-core/src/main/java/com/starrocks/persist/ImageLoader.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+            Path checksumPath = Path.of(imageDir, "v2", Storage.CHECKSUM + "." + imageJournalId);
+            File checksumFile = new File(checksumPath.toUri());
+            if (!checksumFile.exists()) {
+                return;
+            }
+
+            long expectedCheckSum = Long.parseLong(Files.readString(checksumPath));
+            try (CheckedInputStream in = new CheckedInputStream(new FileInputStream(imageFile), new CRC32())) {
+                byte[] bytes = new byte[8192];
+                while (in.read(bytes) != -1) {
```
[the remark this answers is not in the document]
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/persist/ImageLoader.java]: I use the checkedInputStream to judge the checksum at the end. So no pre-check required.

**Review by nshangyiming — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Pull request merged**